### DISCUSSION TITLE
Opentelemetry-api version compatibility matrix

### DISCUSSION THREAD

**Discussion by patschl** [open-telemetry/opentelemetry-java-instrumentation#12754] [Q&A]
Hi!

I was wondering if there is a way to determine which opentelemetry-api SDK versions are compatible or supported. I searched but couldn't find any definitive information.

The most straightforward solution would be to use the same OpenTelemetry SDK version as opentelemetry-java-instrumentation. However, there are cases where the SDK version might be managed by another framework, such as Spring Boot.

Is the only officially supported SDK version the one specified on the release page?

This issue arose when upgrading `io.opentelemetry.instrumentation:opentelemetry-instrumentation-api` to 2.9.0, which broke our instrumentation. The root cause appears to be that the latest Spring Boot BOM (version 3.3.5) currently uses `io.opentelemetry:opentelemetry-api` version 1.37.0, which is not compatible.

Kind regards

**Comment by trask** [comment]
you will need to use the opentelemetry bom to ensure version alignment

check out this note in https://opentelemetry.io/docs/zero-code/java/spring-boot-starter/getting-started/#dependency-management

> Note
Be careful not to mix up the different ways of configuring things with Gradle. For example, don’t use implementation(platform("io.opentelemetry.instrumentation:opentelemetry-instrumentation-bom:2.6.0")) with the io.spring.dependency-management plugin.
